src/libs/fsdrv/lv_fs_memfs.c

```diff
@@ -0,0 +1,233 @@
+ *
+ * lv_fs_path_ex_t mempath;
+ *
+ * This structure can be initialized with the helper function:
+ *
+ * lv_fs_make_path_ex(&mempath, (const uint8_t *) & my_mem_buffer, sizeof(my_mem_buffer));
+ *
+ * Then the "file" can be opened with:
+ *
+ * lv_fs_file_t file;
+ * lv_fs_res_t res = lv_fs_open(&file, (const char *) & mempath, LV_FS_MODE_RD);
+ *
+ * The path object can be used at any place where a file path is required, e.g.:
+ *
+ * lv_font_t* my_font = lv_font_load((const char *) & mempath);
+ *
+ */
+
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+#if LV_USE_FS_MEMFS != '\0'
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+ /**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+//NOTE: although these functions do not do anything, we need them for the file system interface to function correctly
+static void* fs_open(lv_fs_drv_t* drv, const char* path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p);
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br);
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p);
+#if 0
+static lv_fs_res_t fs_write(lv_fs_drv_t* drv, void* file_p, const void* buf, uint32_t btw, uint32_t* bw);
+static void * fs_dir_open(lv_fs_drv_t * drv, const char * path);
+static lv_fs_res_t fs_dir_read(lv_fs_drv_t * drv, void * dir_p, char * fn);
+static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p);
+#endif
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+static lv_fs_drv_t fs_drv; /*A driver descriptor*/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * Register a driver for the File system interface
+ */
+void lv_fs_memfs_init(void)
+{
+    /*---------------------------------------------------
+     * Register the file system interface in LVGL
+     *--------------------------------------------------*/
+
+    /*Add a simple drive to open images*/
+    lv_fs_drv_init(&fs_drv);
+
+    /*Set up fields...*/
+    fs_drv.letter = LV_FS_MEMFS_LETTER;
+    fs_drv.cache_size = UINT32_MAX; //LV_FS_MEMFS_CACHE_SIZE;
+
+    fs_drv.open_cb = fs_open;
+    fs_drv.close_cb = fs_close;
+    fs_drv.read_cb = fs_read;
+    fs_drv.write_cb = NULL; // fs_write;
+    fs_drv.seek_cb = fs_seek;
+    fs_drv.tell_cb = fs_tell;
+
+    fs_drv.dir_close_cb = NULL; // fs_dir_close;
+    fs_drv.dir_open_cb = NULL; // fs_dir_open;
+    fs_drv.dir_read_cb = NULL; // fs_dir_read;
+
+    lv_fs_drv_register(&fs_drv);
+}
+
+void lv_fs_make_path_ex(lv_fs_path_ex_t* path, const uint8_t* buf, uint32_t size)
```

Comment:
By using `const void* buf` we could avoid the cast when this function is called.

Reply:
Changed.